src/JustSaying/AwsTools/MessageHandling/SqsMessagePublisher.cs

```diff
@@ -112,4 +114,121 @@ public InterrogationResult Interrogate()
+                            response.Successful.Count,
+                            "Queue",
+                            request.QueueUrl);
+
+                        foreach (var message in response.Successful)
+                        {
+                            _logger.LogInformation(
+                                "Published message {MessageId} of type {MessageType} to {DestinationType} '{MessageDestination}'.",
+                                message.Id,
+                                message.GetType().FullName,
+                                "Queue",
+                                request.QueueUrl);
+                        }
+                    }
+
+                    if (response.Failed.Count > 0)
+                    {
+                        _logger.LogError(
+                            "Fail to published batch of {MessageCount} to {DestinationType} '{MessageDestination}'.",
+                            response.Failed.Count,
+                            "Queue",
+                            request.QueueUrl);
+
+                        foreach (var message in response.Failed)
+                        {
+                            _logger.LogWarning(
+                                "Fail to published message {MessageId} to {DestinationType} '{MessageDestination}' with error code: {ErrorCode} is error on BatchAPI: {IsBatchAPIError}.",
+                                message.Id,
+                                "Queue",
+                                request.QueueUrl,
+                                message.Code,
+                                message.SenderFault);
+                        }
+                    }
+                }
+            }
+
+            if (MessageBatchResponseLogger != null)
+            {
+                var responseData = new MessageBatchResponse
+                {
+                    SuccessfulMessageIds = response?.Successful.Select(x => x.MessageId),
+                    FailedMessageIds = response?.Failed.Select(x => x.Id),
+                    ResponseMetadata = response?.ResponseMetadata,
+                    HttpStatusCode = response?.HttpStatusCode,
+                };
+
+                MessageBatchResponseLogger(responseData, chuck);
+            }
+        }
+    }
+
+    private SendMessageBatchRequest BuildSendMessageBatchRequest(Message[] message, PublishMetadata metadata)
+    {
+        var request = new SendMessageBatchRequest
+        {
+            QueueUrl = QueueUrl.AbsoluteUri,
+            Entries = message.Select(message =>
+            {
+                var entry = new SendMessageBatchRequestEntry
+                {
+                    Id = message.UniqueKey(),
+                    MessageBody = GetMessageInContext(message)
+                };
+
+                if (metadata?.Delay != null)
+                {
+                    entry.DelaySeconds = (int)metadata.Delay.Value.TotalSeconds;
+                }
```

Comment:
I _think_ you'd be able to simplify this to:

```suggestion
                if (metadata?.Delay is { } delay)
                {
                    entry.DelaySeconds = (int)delay.TotalSeconds;
                }
```